app/src/main/java/br/com/catbag/gifreduxsample/ui/components/FeedComponent.java

```diff
@@ -58,29 +57,33 @@ public FeedComponent(Context context, AttributeSet attrs, int defStyleAttr) {
 
     @Override
     public void view() {
-        final List<Gif> gifs = mGifs;
         Recycler.view(() -> {
             if (mLayoutManager == null) {
                 mLayoutManager = new LinearLayoutManager(getContext(),
                         LinearLayoutManager.VERTICAL, false);
             }
             Recycler.layoutManager(mLayoutManager);
             Recycler.hasFixedSize(true);
-            Recycler.adapter(Recycler.Adapter.simple(gifs, (viewHolder) -> {
-                renderGifView(gifs.get(viewHolder.getAdapterPosition()));
-            }));
+            if (mGifsAdapter == null) {
+                mGifsAdapter = gifsAdapter(mGifs, (gif) -> renderGifView(gif));
+                Recycler.adapter(mGifsAdapter);
+            }
+            else if (!mGifs.equals(mGifsAdapter.getGifs())) {
```

Comment:
This else if pattern looks weird to me.
I prefer the common and straightforward way:

```java
if (....) {
   //...
} else if (.....) {
   //...
}

But it's ok! However we need to define a pattern to be used over all code, since I believe that there is different else if style along the code.